catroid/src/test/java/org/catrobat/catroid/stage/EmbroideryActorCircleTest.java

```diff
@@ -0,0 +1,108 @@
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.catrobat.catroid.stage;
+
+import com.badlogic.gdx.graphics.g2d.Batch;
+import com.badlogic.gdx.graphics.glutils.ShapeRenderer;
+
+import org.catrobat.catroid.embroidery.EmbroideryPatternManager;
+import org.catrobat.catroid.embroidery.StitchPoint;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.mockito.verification.VerificationMode;
+
+import java.util.List;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import static java.util.Arrays.asList;
+
+@RunWith(Parameterized.class)
+public class EmbroideryActorCircleTest {
+	@Parameters(name = "{0}")
+	public static Iterable<Object[]> data() {
+		return asList(new Object[][] {
+				{"Test JumpPoint after ChangePoint", asList(getChangePointMock(), getJumpPointMock()), false},
+				{"Test Connecting Points", asList(getConnectingPointMock(), getConnectingPointMock()), true},
+				{"Test Changing Points", asList(getChangePointMock(), getChangePointMock()), false},
+				{"Test ChangePoint after JumpPoint", asList(getJumpPointMock(), getChangePointMock()), false},
+				{"Test JumpPoint after ConnectingPoint", asList(getConnectingPointMock(), getJumpPointMock()), true},
+				{"Test Jumping Points", asList(getJumpPointMock(), getJumpPointMock()), false},
+		});
+	}
+
+	@Parameter
+	public String name;
+
+	@Parameter(1)
+	public List<StitchPoint> stitchPoints;
+
+	@Parameter(2)
+	public boolean firstPointDrawn;
+
+	private EmbroideryActor embroideryActorSpy;
+
+	@Before
+	public void setUp() throws Exception {
+		EmbroideryPatternManager embroideryPatternManager = mock(EmbroideryPatternManager.class);
+		when(embroideryPatternManager.getEmbroideryPatternList()).thenReturn(stitchPoints);
+		ShapeRenderer renderer = mock(ShapeRenderer.class);
+		embroideryActorSpy = spy(new EmbroideryActor(1.0f, embroideryPatternManager, renderer));
+	}
+
+	@Test
+	public void testDraw() {
+		embroideryActorSpy.draw(mock(Batch.class), 1.0F);
+
+		VerificationMode mode = firstPointDrawn ? times(1) : never();
+		verify(embroideryActorSpy, mode).drawCircle(stitchPoints.get(0));
```

Comment:
Please verify (maybe in an extra test method) that the second stitch-point is always drawn.